### PR TITLE
Uploading the file on AnyFetch

### PR DESCRIPTION
**Pull request by Antonhansel**

### CODE REVIEW

File: lib/helpers/upload.js [hunk 1]
```diff
@@ -18,14 +33,58 @@ module.exports = function upload(file, anyfetchClient, accessToken, cb) {
...
     },
-    function processFolderContent(content, cb) {
-      console.log(content);
-      cb(null);
+    function handleMetada(metadata, cb) {
+      metadata = metadata.body;
+      // Let's build the file address for the user
+      document.identifier = 'https://app.box.com/files/0/f/' + metadata.parent.id + '/1/f_' + metadata.id;
+      document.actions.show = document.identifier;
+      // do not download the filenamee when it is scheduled for deletion or if it is bigger than 50Mb
```
Comment (Neamar) [inline lib/helpers/upload.js]: filename

Comment (Neamar) [inline lib/helpers/upload.js]: in fact, "do not download the file"

File: lib/helpers/upload.js [hunk 2]
```diff
@@ -18,14 +33,58 @@ module.exports = function upload(file, anyfetchClient, accessToken, cb) {
...
+        fileConfig.knownLength = metadata.size;
+        document.creation_date = metadata.created_at;
+        document.modification_date = metadata.modified_at;
+        document.metadata = {
+          title: metadata.name
+        };
+        cb(null);
+      }
+      else {
+        cb('Error: File will be soon deleted');
```
Comment (Neamar) [inline lib/helpers/upload.js]: cb(new Errror("File will ..."))

File: lib/helpers/upload.js [hunk 3]
```diff
@@ -18,14 +33,58 @@ module.exports = function upload(file, anyfetchClient, accessToken, cb) {
...
+      }
+    },
+    function getFileUrl(cb) {
+      request(apiUrl)
+        .get(url + '/content/')
+        .set('Content-Type', 'application/x-www-form-urlencoded')
+        .set('Authorization', 'Bearer ' + accessToken)
+        .expect(200)
+        .end(function(err, res) {
```
Comment (Neamar) [inline lib/helpers/upload.js]: waterfall ;)

Comment (Antonhansel) [inline lib/helpers/upload.js]: I actually cant use waterfall in this case.
The response can either be a 302 or 202 as documented in the box api documentation.
https://developers.box.com/docs/#files-download-a-file

File: lib/workers.js [hunk 4]
```diff
@@ -2,7 +2,7 @@
 
 var log = require('anyfetch-provider').log;
 
-// var uploadContact = require('./helpers/upload.js');
+var uploadContact = require('./helpers/upload.js');
```
Comment (Neamar) [inline lib/workers.js]: contact?

File: lib/workers.js [hunk 5]
```diff
@@ -12,8 +12,7 @@ module.exports.addition = function additionQueueWorker(job, cb) {
     name: 'addition',
     identifier: job.task.identifier
   }, "Uploading");
-  // uploadContact(job.task, job.anyfetchClient, job.serviceData.accessToken, cb);
-  cb(null);
+  uploadContact(job.task, job.anyfetchClient, job.serviceData.access_token, cb);
```
Comment (Neamar) [inline lib/workers.js]: contact?

**Comment by Neamar** [comment]
Also, we're missing a test ;) (next PR)

File: lib/helpers/upload.js [hunk 6]
```diff
@@ -18,14 +33,58 @@ module.exports = function upload(file, anyfetchClient, accessToken, cb) {
...
+        cb(new Error('File will be soon deleted'));
+      }
+    },
+    function getFileUrl(cb) {
+      request(apiUrl)
+        .get(url + '/content/')
+        .set('Content-Type', 'application/x-www-form-urlencoded')
+        .set('Authorization', 'Bearer ' + accessToken)
+        .expect(302)
+        .end(function(err, res) {
```
Comment (Neamar) [inline lib/helpers/upload.js]: waterfall

File: lib/helpers/upload.js [hunk 7]
```diff
@@ -18,14 +33,58 @@ module.exports = function upload(file, anyfetchClient, accessToken, cb) {
...
+          if(res) {
+            if(res.statusCode === 302) {
+              return cb(null, res.header.location);
+            }
+            else if(res.statusCode === 202) {
+              return cb('Testing the retry after header:' + res.header['Retry-After']);
```
Comment (Neamar) [inline lib/helpers/upload.js]: cb(new Error

File: lib/helpers/upload.js [hunk 8]
```diff
@@ -18,14 +33,58 @@ module.exports = function upload(file, anyfetchClient, accessToken, cb) {
...
     }
-    ], function(err) {
-      console.log('Error in upload:' + err);
-      cb(null);
-    });
-
-  // anyfetchClient.postDocument(formattedContact, rarity.slice(1, cb));
+  ], function(err) {
+    if(err) {
+      return log.error(err);
```
Comment (Neamar) [inline lib/helpers/upload.js]: send it to opbeat

Comment (Neamar) [inline lib/helpers/upload.js]: return cb(err)

**Pull request merged**